ui/src/components/common/RequestCustodianRelationship.tsx

```diff
@@ -39,27 +37,11 @@ const RequestCustodianRelationship: React.FC<Props> = ({ role, custodianRelation
             !requestCustodians.includes(custodian.contractData.custodian) &&
             !relationshipCustodians.includes(custodian.contractData.custodian));
 
-    const requestCustodianRelationship = async (custodianId: string) => {
+    const requestCustodianRelationship = async (custodianIds: string[]) => {
         const key = wrapDamlTuple([operator, party]);
 
-        const args = { custodian: custodianId };
-
-        switch(role) {
-            case MarketRole.InvestorRole:
-                await ledger.exerciseByKey(Investor.Investor_RequestCustodianRelationship, key, args);
-                break;
-            case MarketRole.IssuerRole:
-                await ledger.exerciseByKey(Issuer.Issuer_RequestCustodianRelationship, key, args);
-                break;
-            case MarketRole.BrokerRole:
-                await ledger.exerciseByKey(Broker.Broker_RequestCustodianRelationship, key, args);
-                break;
-            case MarketRole.ExchangeRole:
-                await ledger.exerciseByKey(Exchange.Exchange_RequestCustodianRelationship, key, args);
-                break;
-            default:
-                throw new Error(`The role '${role}' can not request a custodian relationship.`);
-        }
```

Comment:
nice 👍

Reply:
yep yep, my typescript chops have sort of improved since this was added. I wanted to do it this way the first time, but I couldn't figure out how to type the choice param generically (the VS Code type hint was some crazy nonsense like `Choice<blahblah, blahblahblah, ...>`)

Now I know you can just do `typeof choiceVar` and it evaluates to that crazy nonsense for you!

Reply:
good to know!